test/zod-collection.test.ts

```diff
@@ -25,6 +26,23 @@ test('Documents should be parsed correctly on insert', async () => {
     zodCol.insertOne({ a: 0, b: 'string', numbers: [0] })
   ).resolves.toBeTruthy()
 
+  const id = new ObjectId()
+  const { insertedId } = await zodCol.insertOne({
+    a: 0,
+    b: 'string',
+    numbers: [0],
+    _id: id,
+  })
+  expect(insertedId.equals(id)).toBeTruthy()
+
+  expect(() =>
+    zodCol.insertOne({ a: 'string', b: 'string', numbers: [0], _id: null })
+  ).toThrow()
+
+  expect(() =>
+    zodCol.insertOne({ a: 'string', b: 'string', numbers: [0], _id: 'idStr' })
+  ).toThrow()
+
```

Comment:
This will be more clear in a separate test "Should allow inserting documents with `_id`"